lib/cloud/awsconfig/awsconfig.go

```diff
@@ -324,6 +331,14 @@ func WithBaseCredentialsProvider(baseCredentialsProvider aws.CredentialsProvider
 	}
 }
 
+// WithEC2IMDSRegion indicates that clients should determine their region from
+// EC2 instance metadata if available.
+func WithEC2IMDSRegion() OptionsFn {
```

Comment:
is there any side effect with this enabled? like does it time out or anything on non-EC2 environments? If not, any downside always enabling it?

Reply:
It will time out on non-EC2.